src/sonic-yang-models/yang-models/sonic-hash.yang

```diff
@@ -0,0 +1,67 @@
+module sonic-hash {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-hash";
+    prefix hash;
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    description "HASH YANG Module for SONiC OS";
+
+    revision 2022-09-05 {
+        description "First Revision";
+    }
+
+    typedef hash-field {
```

Comment:
you have defined hash-field in sonic-types.yang

Reply:
@ganglyu correct .A subset is used to enable only fields which are relevant for this feature

Reply:
Both sonic-pbh.yang and sonic-hash.yang use hash-field, and the fields in hash-field are different.
But your implementation does not align, sonic-pbh.yang uses hash-field defined in sonic-types.yang, and overwrite this field, sonic-hash.yang defines a new hash-field.
It seems that both sonic-phb.yang and sonic-hash.yang can define hash-field for them self?

Reply:
Good point, then you should do the same for sonic-hash.yang

Reply:
@ganglyu it is already done:

**COMMON:**
```
typedef hash-field {
    description "Represents native hash field";
    type enumeration {
        enum IN_PORT;
        enum DST_MAC;
        enum SRC_MAC;
        enum ETHERTYPE;
        enum VLAN_ID;
        enum IP_PROTOCOL;
        enum DST_IP;
        enum SRC_IP;
        enum L4_DST_PORT;
        enum L4_SRC_PORT;
        enum INNER_DST_MAC;
        enum INNER_SRC_MAC;
        enum INNER_ETHERTYPE;
        enum INNER_IP_PROTOCOL;
        enum INNER_DST_IP;
        enum INNER_DST_IPV4;
        enum INNER_DST_IPV6;
        enum INNER_SRC_IP;
        enum INNER_SRC_IPV4;
        enum INNER_SRC_IPV6;
        enum INNER_L4_DST_PORT;
        enum INNER_L4_SRC_PORT;
    }
}
```

**HASH:**
```
import sonic-types {
    prefix stypes;
}

...

typedef hash-field {
    description "Represents native hash field";
    type stypes:hash-field {
        enum IN_PORT;
        enum DST_MAC;
        enum SRC_MAC;
        enum ETHERTYPE;
        enum VLAN_ID;
        enum IP_PROTOCOL;
        enum DST_IP;
        enum SRC_IP;
        enum L4_DST_PORT;
        enum L4_SRC_PORT;
        enum INNER_DST_MAC;
        enum INNER_SRC_MAC;
        enum INNER_ETHERTYPE;
        enum INNER_IP_PROTOCOL;
        enum INNER_DST_IP;
        enum INNER_SRC_IP;
        enum INNER_L4_DST_PORT;
        enum INNER_L4_SRC_PORT;
    }
}
```

**PBH:**
```
import sonic-types {
    prefix stypes;
}

...

typedef hash-field {
    description "Represents native hash field";
    type enumeration {
    type stypes:hash-field {
        enum INNER_IP_PROTOCOL;
        enum INNER_L4_DST_PORT;
        enum INNER_L4_SRC_PORT;
        enum INNER_DST_IPV4;
        enum INNER_SRC_IPV4;
        enum INNER_DST_IPV6;
        enum INNER_SRC_IPV6;
    }
}
```